StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/STPElementsSessionTest.swift

```diff
@@ -113,6 +113,77 @@ class STPElementsSessionTest: XCTestCase {
         }))
     }
 
+    func testFailedCPMParsing() {
+        // If STPElementsSession decodes a dict...
+        var elementsSessionJson = STPTestUtils.jsonNamed("ElementsSession")!
+        // ...that contains unparseable custom_payment_method_data
+        elementsSessionJson["custom_payment_method_data"] = [
+            "this dict doesn't match the expected shape": "and will fail to parse",
+        ]
+        let elementsSession = STPElementsSession.decodedObject(fromAPIResponse: elementsSessionJson)!
+        // ...it should successfully decode...
+        XCTAssertNotNil(elementsSession)
+        // ...with an empty `customPaymentMethods` property
+        XCTAssertTrue(elementsSession.customPaymentMethods.isEmpty)
+        // ...and send a failure analytic
+        let analyticEvents = STPAnalyticsClient.sharedClient._testLogHistory
+        XCTAssertTrue(analyticEvents.contains(where: { dict in
+            (dict["event"] as? String) == STPAnalyticEvent.paymentSheetElementsSessionCPMLoadFailed.rawValue
+        }))
+    }
+
+    func testSuccessCPMParsing() {
+        // If STPElementsSession decodes a dict...
+        var elementsSessionJson = STPTestUtils.jsonNamed("ElementsSession")!
+
+        let customPaymentMethodsArray: [[String: Any]] = [
+            [
+                "logo_url": "stripe.com",
```

Comment:
should logo_url be a fully qualified url: etc:  https://  -- i'd expect this fail?

Reply:
Added https!